.github/workflows/build-deploy-pudl.yml

```diff
@@ -33,29 +33,46 @@ jobs:
         uses: actions/checkout@v4
         with:
           ref: ${{ env.BUILD_REF }}
+          fetch-depth: 0
+
+      - name: Skip the build if no changes since the last successful nightly build.
+        run: |
+          CURRENT_COMMIT=$(git rev-parse HEAD)
+          NIGHTLY_COMMIT=$(git rev-parse origin/nightly)
+          if [[ "$CURRENT_COMMIT" == "$NIGHTLY_COMMIT" ]]; then
+            echo "::notice::No changes since last successful nightly build. Skipping."
+            echo "SKIP_BUILD=true" >> $GITHUB_ENV
+            exit 0
+          fi
```

Comment:
See if anything has actually changed since the last nightly build, and if not, skip doing another one.